### ISSUE TITLE
adding GAP (with or without packages) fails

### ISSUE THREAD

**Issue by kalmarek** [oscar-system/GAP.jl#304]
```julia
julia> ENV["GAP_INSTALL_PACKAGES"] = "no"
"no"

(gap) pkg> add GAP
 Resolving package versions...
 Installed GAP ─ v0.2.2
  Updating `/tmp/gap/Project.toml`
  [c863536a] + GAP v0.2.2
  Updating `/tmp/gap/Manifest.toml`
  [c863536a] + GAP v0.2.2
  [cd4130c4] + GAPTypes v1.0.0
  [2a0f44e3] + Base64 
  [ade2ca70] + Dates 
  [b77e0a4c] + InteractiveUtils 
  [76f85450] + LibGit2 
  [8f399da3] + Libdl 
  [56ddb016] + Logging 
  [d6f4376e] + Markdown 
  [44cfe95a] + Pkg 
  [de0858da] + Printf 
  [3fa0cd96] + REPL 
  [9a3f8284] + Random 
  [ea8e919c] + SHA 
  [9e88b42a] + Serialization 
  [6462fe0b] + Sockets 
  [cf7118a7] + UUIDs 
  [4ec0a83e] + Unicode 
  Building GAP → `~/.julia/packages/GAP/W20PN/deps/build.log`
┌ Error: Error building `GAP`, showing the last 100 of log: 
│    C       src/modules_builtin.c => obj/src/modules_builtin.lo
│    C       src/objcftl.c => obj/src/objcftl.lo
│    C       src/objects.c => obj/src/objects.lo
│    CXX     src/objfgelm.cc => obj/src/objfgelm.lo
│    CXX     src/objpcgel.cc => obj/src/objpcgel.lo
│    C       src/objset.c => obj/src/objset.lo
│    C       src/opers.c => obj/src/opers.lo
│    CXX     src/permutat.cc => obj/src/permutat.lo
│    C       src/plist.c => obj/src/plist.lo
│    CXX     src/pperm.cc => obj/src/pperm.lo
│    C       src/precord.c => obj/src/precord.lo
│    C       src/profile.c => obj/src/profile.lo
│    C       src/range.c => obj/src/range.lo
│    C       src/rational.c => obj/src/rational.lo
│    C       src/read.c => obj/src/read.lo
│    C       src/records.c => obj/src/records.lo
│    C       src/saveload.c => obj/src/saveload.lo
│    C       src/scanner.c => obj/src/scanner.lo
│    C       src/sctable.c => obj/src/sctable.lo
│    C       src/set.c => obj/src/set.lo
│    C       src/stats.c => obj/src/stats.lo
│    C       src/streams.c => obj/src/streams.lo
│    C       src/stringobj.c => obj/src/stringobj.lo
│    C       src/syntaxtree.c => obj/src/syntaxtree.lo
│    C       src/sysfiles.c => obj/src/sysfiles.lo
│    C       src/sysmem.c => obj/src/sysmem.lo
│    C       src/system.c => obj/src/system.lo
│    C       src/tietze.c => obj/src/tietze.lo
│    CXX     src/trans.cc => obj/src/trans.lo
│    C       src/vars.c => obj/src/vars.lo
│    C       src/vec8bit.c => obj/src/vec8bit.lo
│    C       src/vecffe.c => obj/src/vecffe.lo
│    C       src/vecgf2.c => obj/src/vecgf2.lo
│    C       src/vector.c => obj/src/vector.lo
│    C       src/weakptr.c => obj/src/weakptr.lo
│    C       src/julia_gc.c => obj/src/julia_gc.lo
│    LINK    gap
│    LINK    libgap.la
│   Updating registry at `~/.julia/registries/General`
│   Updating git-repo `https://github.com/JuliaRegistries/General.git`
   Resolving package versions...
│  Installed DocSeeker ─────────── v0.3.2
│  Installed Mux ───────────────── v0.7.1
│  Installed MultivariateMoments ─ v0.2.4
│  Installed DataStructures ────── v0.17.9
│  Installed JuliaInterpreter ──── v0.7.8
│  Installed FileIO ────────────── v1.2.1
│  Installed Cassette ──────────── v0.3.0
│  Installed PlotUtils ─────────── v0.6.2
│  Installed ErrorfreeArithmetic ─ v0.4.1
│  Installed ForwardDiff ───────── v0.10.9
│  Installed ZMQ ───────────────── v1.2.0
│  Installed ZeroMQ_jll ────────── v4.3.1+0
│  Installed GeometryTypes ─────── v0.7.7
│  Installed ColorTypes ────────── v0.9.1
│  Installed Colors ────────────── v0.11.2
│  Installed Blink ─────────────── v0.12.1
│   Updating `~/.julia/environments/v1.3/Project.toml`
│   [cd4130c4] + GAPTypes v1.0.0
│   Updating `~/.julia/environments/v1.3/Manifest.toml`
│   [ad839575] ↑ Blink v0.12.0 ⇒ v0.12.1
│   [7057c7e9] ↑ Cassette v0.2.6 ⇒ v0.3.0
│   [3da002f7] ↑ ColorTypes v0.9.0 ⇒ v0.9.1
│   [5ae59095] ↑ Colors v0.11.0 ⇒ v0.11.2
│   [864edb3b] ↑ DataStructures v0.17.6 ⇒ v0.17.9
│   [33d173f1] ↑ DocSeeker v0.3.1 ⇒ v0.3.2
│   [90fa49ef] ↑ ErrorfreeArithmetic v0.4.0 ⇒ v0.4.1
│   [5789e2e9] ↑ FileIO v1.2.0 ⇒ v1.2.1
│   [f6369f11] ↑ ForwardDiff v0.10.8 ⇒ v0.10.9
│   [cd4130c4] + GAPTypes v1.0.0
│   [4d00f742] ↑ GeometryTypes v0.7.6 ⇒ v0.7.7
│   [c8e1da08] - IterTools v1.3.0
│   [aa1ae85d] ↑ JuliaInterpreter v0.7.6 ⇒ v0.7.8
│   [50d2b5c4] ↑ Lazy v0.13.2 ⇒ v0.14.0
│   [f4abf1af] ↑ MultivariateMoments v0.2.3 ⇒ v0.2.4
│   [a975b10e] ↑ Mux v0.7.0 ⇒ v0.7.1
│   [995b91a9] ↑ PlotUtils v0.6.1 ⇒ v0.6.2
│   [c2297ded] ↑ ZMQ v1.0.0 ⇒ v1.2.0
│   [8f1865be] + ZeroMQ_jll v4.3.1+0
│   Building Blink → `~/.julia/packages/Blink/4FM8Y/deps/build.log`
│ Configuring JuliaInterface
│ Using settings from /home/kalmar/.julia/packages/GAP/W20PN/gap/sysinfo.gap
│ cp: cannot create regular file './Makefile.gappkg': Permission denied
│ ERROR: LoadError: failed process: Process(`./configure /home/kalmar/.julia/packages/GAP/W20PN/gap`, ProcessExited(1)) [1]
│ 
│ Stacktrace:
│  [1] pipeline_error at ./process.jl:525 [inlined]
│  [2] #run#565(::Bool, ::typeof(run), ::Cmd) at ./process.jl:440
│  [3] run(::Cmd) at ./process.jl:438
│  [4] top-level scope at /home/kalmar/.julia/packages/GAP/W20PN/deps/build.jl:77
│  [5] include at ./boot.jl:328 [inlined]
│  [6] include_relative(::Module, ::String) at ./loading.jl:1105
│  [7] include(::Module, ::String) at ./Base.jl:31
│  [8] include(::String) at ./client.jl:424
│  [9] top-level scope at none:5
│ in expression starting at /home/kalmar/.julia/packages/GAP/W20PN/deps/build.jl:77
│ gap_root = /home/kalmar/.julia/packages/GAP/W20PN/gap
│ extra_gap_root = /home/kalmar/.julia/packages/GAP/W20PN/
│ install_gap = true
│ Installing GAP ...
│ Compiling JuliaInterface and JuliaExperimental ...
│ 
│ Full log at /home/kalmar/.julia/packages/GAP/W20PN/deps/build.log
└ @ Pkg.Operations /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.3/Pkg/src/backwards_compatible_isolation.jl:649
```

**Comment by fingolfin** [comment]
@kalmarek thanks, i got a report on the same issue by @sebastianpos today. I don't quite understand why Travis doesn't catch this, so besides fixing this, I'll have to make sure Travis catches it next time around.

**Comment by fingolfin** [comment]
Could either of you perhaps send me the full `build.log` (resp. attach it here)

**Comment by kalmarek** [comment]
actually I've just tried with `GAP_INSTALL_PACKAGES=yes`
https://gist.github.com/kalmarek/5e91df6f820417e506c2a6e8b9ce8650

**Comment by fingolfin** [comment]
So this regression seems to be caused by a change in Julia (in `PKg.jl`, I guess?) which now does a `chmod a-w` on all source files. It's still not clear to me why this doesn't show up on Travis.

**Comment by kalmarek** [comment]
I'll try this tomorrow, thanks! 
(for tracking this while the hell broke on the email ;-)

**Comment by fingolfin** [comment]
I made a PR, if it works for you, I'll make a new release of GAP.jl ASAP

**Comment by kalmarek** [comment]
fixed by #307
